R/beta_binomial_ab_test.R

```diff
@@ -77,18 +53,56 @@ beta_binomial_ab_test <- function(y, n,
     risk[g]  <- mean(loss)
   }
 
-  return(list(y = y,
-              n = n,
-              risk = risk,
-              winner = groups[risk < tolerance],
-              stop.test = min(risk) < tolerance,
-              tolerance = tolerance,
-              prob.winning = prob.winning,
-              posterior.mean = posterior.mean,
-              ci = ci,
-              conf.level = conf.level,
-              posterior_parameters = data.frame(alpha = alpha,
-                                                beta = beta),
-              groups = groups)
+  return(
+    structure(
+      list(y = y,
+           n = n,
+           risk = risk,
+           winner = groups[risk < tolerance],
+           stop.test = min(risk) < tolerance,
+           tolerance = tolerance,
+           prob.winning = prob.winning,
+           posterior.mean = posterior.mean,
+           ci = ci,
+           conf.level = conf.level,
+           posterior_parameters = data.frame(alpha = alpha,
+                                             beta = beta),
+           groups = groups),
+      class = "beta_binomial_ab_test")
+  )
+}
+
+plot.beta_binomial_ab_test <- function(x,
+                                       plot.limits = c(0, 1),
+                                       plot.labels = NULL,
+                                       plot.name = "") {
+  ngroups <- length(x$y)
+
+  d <- expand.grid(group = seq(1, ngroups, 1),
+                   input = seq(0, 1, 0.001))
+  d$output <- mapply(FUN = dbeta,
+                     x = d$input,
+                     shape1 = x$posterior_parameters$alpha,
+                     shape2 = x$posterior_parameters$beta)
+
+  if(is.null(plot.labels)) {
+    plot.labels <- 1:ngroups
+  }
+
+  print(
+    ggplot2::ggplot(d,
```

Comment:
Accessing functions like `ggplot` explicitly is a good idea, but for `ggplot2` specifically, we might want to `require` or add to `DEPENDS` since it's used so much.